### PR TITLE
Testing 6

### PR DESCRIPTION
**Pull request by rahul799**
This PR will test 
Different mixes of R vs other files, including confirming that it does nothing when no R files are changed

### CODE REVIEW

File: models/gday/R/model2netcdf.GDAY.R [hunk 1]
```diff
@@ -23,101 +23,100 @@
...
     ## Setup outputs for netCDF file in appropriate units
     output <- list()
-    
+
     ## standard variables: C-Fluxes
     output[[1]] <- (sub.GDAY.output[, "auto_resp"] * THA_2_KG_M2) / timestep.s
     output[[2]] <- (sub.GDAY.output[, "hetero_resp"] * THA_2_KG_M2) / timestep.s
     output[[3]] <- (sub.GDAY.output[, "auto_resp"] + sub.GDAY.output[, "hetero_resp"] *
-                      THA_2_KG_M2) / timestep.s
+      THA_2_KG_M2) / timestep.s
```
Comment (infotroph) [inline models/gday/R/model2netcdf.GDAY.R:58]: It looks as styler doesn't believe in aligning wrapped expressions with their opening paren!

I'm not inclined to fight that and would probably sidestep the issue by reflowing this block as

```
output[[3]] <- (
    sub.GDAY.output[, "auto_resp"]
  + sub.GDAY.output[, "hetero_resp"] * THA_2_KG_M2
  ) / timestep.s
```

... Which now that I type it that way, looks to me like a bug! I wonder if this was actually supposed to be (auto + hetero)*THA_2_KG_M2 / timestep ?

Comment (rahul799) [reply models/gday/R/model2netcdf.GDAY.R:58]: I am not sure :confused:

Comment (infotroph) [reply models/gday/R/model2netcdf.GDAY.R:58]: Sorry, this wasn't clearly worded nor was it the right PR to post it in -- I was just musing about a corner case in styler; no changes needed in the workflow.

**Comment by infotroph** [comment]
The next invocation of the style command will find only `.prettierignore` changed. Since no R files have changed, there's nothing for styler to do. Let's see what happens!

**Comment by infotroph** [comment]
Answer, styler [runs](https://github.com/infotroph/pecan-pr-sandbox/actions/runs/149057073), finds no changes to make, and exits reporting success. Nothing is committed. 

The only downside I see here is that you need to go check the actions tab to tell whether the run is still in progress or completed without finding any changes. But then again most people won't be requesting a styler run when there's nothing to style, so this is probably not a big deal.

**Comment by infotroph** [comment]
The next invocation of the styler command will find no R files changed, but should notice and fix one misedited Rd file. Let's see what happens!

**Comment by infotroph** [comment]
Answer: Styler runs but doesn't commit the change! It took me a while to track down why:

* Issue 1: Initial list of changed files includes the changed base/utils/PEcAn.Rd, but only *.R and *.Rmd are passed in to files_to_style.txt. files_to_style.txt is the only artifact passed to the check step, so the check step never gets the information that base/utils needs to be Roxygenized. To fix this, the check step should be given the full list of changed files.

* Issue 2: But In this case, the process of checking `models/fates` and `models/gday` prompts Make to re-Roxygenize `base/utils` anyway! But the change isn't committed because the command    

    ```
    if [[ $(git diff --name-only --cached) != "" ]]; then  git commit -m 'automated documentation update' ; fi
    ```

    returns `[[: not found`. To fix this, we should either use `[` instead of the bashism `[[`, or explicitly execute the command in bash rather than the default shell.

**Comment by infotroph** [comment]
The next invocation tests again after fixing issue 2 (`[[: not found`) from above. If it's fixed correctly, the next commit should revert the changes to PEcAn.Rd.

Note also that each time we invoke styler it looks at the set of all files changed in this pull request, *not* just the files changed in the most recent commit like I was assuming above. But of course the style changes to *.R have already been committed by previous styler invocations, so we should only see changes in the newly touched files.

**Comment by infotroph** [comment]
/style

(Apparently I didn't actually push the [[ fix before)

**Comment by infotroph** [comment]
Result: In this case the mangled Rd file gets fixed, but by accident (because utils gets rebuilt when `make .doc/models/fates` runs) rather than because of the change. This should be fixed eventually, but "change to Rd file with no corresponding change to R files in the same package" is a weird enough corner case that it needn't stop us from merging this now.

**Pull request closed without merging**